### PR TITLE
Remove SM PLatform credentials support

### PR DESCRIPTION
**Pull request by piotrmiskiewicz**
**Description**

Changes proposed in this pull request:
- Remove SM platform credentials support

**Related issue(s)**

### CODE REVIEW

File: components/kyma-environment-broker/internal/storage/encrypt.go [hunk 1]
```diff
@@ -113,32 +89,17 @@ func (e *Encrypter) EncryptSMCreds(pp *internal.ProvisioningParameters) error {
...
-		if creds.Password != "" {
-			password, err = e.Decrypt([]byte(creds.Password))
-			if err != nil {
-				return errors.Wrap(err, "while decrypting password")
-			}
-		}
-	}
+	var clientID, clientSecret []byte
+
 	if pp.ErsContext.SMOperatorCredentials != nil {
```
Comment (wozniakjan) [inline components/kyma-environment-broker/internal/storage/encrypt.go:103]: nit: I think this check is no longer necessary because on line 97 KEB will short-circuit
https://github.com/kyma-project/control-plane/blob/b5fd6f562b159bce522b9f83a3d7e9216ca0e493/components/kyma-environment-broker/internal/storage/encrypt.go#L97-L99

File: components/kyma-environment-broker/internal/storage/encrypt.go [hunk 2]
```diff
@@ -60,36 +60,12 @@ func (e *Encrypter) Decrypt(obj []byte) ([]byte, error) {
...
-			Credentials: internal.ServiceManagerCredentials{
-				BasicAuth: internal.ServiceManagerBasicAuth{
-					Password: string(password),
-					Username: string(username),
-				},
-			},
-		}
-	}
+
 	if pp.ErsContext.SMOperatorCredentials != nil {
```
Comment (wozniakjan) [inline components/kyma-environment-broker/internal/storage/encrypt.go:69]: nit: also here I think we can drop this due to
https://github.com/kyma-project/control-plane/blob/b5fd6f562b159bce522b9f83a3d7e9216ca0e493/components/kyma-environment-broker/internal/storage/encrypt.go#L63-L65

**Review by wozniakjan — APPROVED** [review]
/lgtm
/approve
/hold

feel free to unhold at your convenience, the extra `if`s are harmless

File: components/kyma-environment-broker/internal/storage/encrypt.go (hunk 2 above)
Comment (piotrmiskiewicz) [reply components/kyma-environment-broker/internal/storage/encrypt.go]: applied

**Review by wozniakjan — DISMISSED** [review]

File: resources/kcp/values.yaml [hunk 3]
```diff
@@ -158,11 +158,12 @@ global:
...
       - name: "LOG_ISTIO_OPERATOR"
         value: "false"
-      - name: "DEBUG_LOGGING_FOR_SPECIFIC_OPERATIONS"
-        value: "true"
   component_reconcilers:
     enabled: false
     authentication: true
+  oathkeeper:
+    host: ory-oathkeeper-proxy.kyma-system.svc.cluster.local
+    port: 4455
```
Comment (wozniakjan) [inline resources/kcp/values.yaml:166]: seems unrelated to SM platform, added this on purpose?

Comment (piotrmiskiewicz) [reply resources/kcp/values.yaml]: fixed (mistake when conflicts were resovled)

**Review by wozniakjan — DISMISSED** [review]
/lgtm
/approve

**Review by wozniakjan — APPROVED** [review]
/lgtm
/approve
